src/joystick/virtual/SDL_virtualjoystick.c

```diff
@@ -142,6 +142,10 @@ SDL_JoystickID SDL_JoystickAttachVirtualInner(const SDL_VirtualJoystickDesc *des
         SDL_InvalidParamError("desc");
         return 0;
     }
+    if (desc->version != sizeof(*desc)) {
```

Comment:
What compatibility does SDL3 promise?
Should a binary built against a new SDL3 (but not using newer SDL3 functionality) work with an older SDL3 library?
If that's the case, then I think this should be `if (desc->version < sizeof(*desc)) { FAIL }`

Reply:
Well, we don't want it to accept _any_ data less than; if you hand me a struct with three bytes, that's probably a bug.  :)

Reply:
In general we don't make any promises about running on an older SDL, but in this case it makes sense to allow it. (Ryan and I discussed this case in passing as well.)